### PR TITLE
Handle conditional access operator in CompareSymbolsCorrectlyFix

### LINKED ISSUE #4337: Use symbols correctly: code fix bug with null-conditional access operator
Caught this while applying the code fix to:

https://github.com/dotnet/roslyn-analyzers/blob/91b109f2f69496b76e7a740a4547a9d9b9a12c0b/src/Utilities/Compiler/Extensions/ISymbolExtensions.cs#L296

It was transformed into:

```csharp
if (self? SymbolEqualityComparer.Default.Equals(self, overload) == true)
```

### PR DESCRIPTION
**Pull request by Evangelink**
Fixes #4337 and overrides #4341

@Youssef1313 I couldn't push to your branch so I am creating this separate PR, you are still author of all the work you have done so it should be good to close the other PR and move forward with this one. If you prefer to just cherry-pick commits, let me know.

@mavasani Ready for review

### CODE REVIEW

File: src/Microsoft.CodeAnalysis.Analyzers/CSharp/MetaAnalyzers/Fixers/CSharpCompareSymbolsCorrectlyFix.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+using System.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.Fixers;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.Analyzers.MetaAnalyzers.Fixers
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(CompareSymbolsCorrectlyFix)), Shared]
+    public sealed class CSharpCompareSymbolsCorrectlyFix : CompareSymbolsCorrectlyFix
```
Comment (Evangelink) [inline src/Microsoft.CodeAnalysis.Analyzers/CSharp/MetaAnalyzers/Fixers/CSharpCompareSymbolsCorrectlyFix.cs:13]: We could get rid of this when we move to a newer version of roslyn where we have the `ConditionalAccessExpression` on the `SyntaxGenerator`.

**Comment by Youssef1313** [comment]
Thanks for completing the fix! :heart:

**Comment by Evangelink** [comment]
@mavasani Could you review when you have some time. Thanks.

**Comment by Evangelink** [comment]
Ping @mavasani

File: src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/Fixers/CompareSymbolsCorrectlyFix.cs [hunk 2]
```diff
@@ -65,25 +68,68 @@ private static async Task<Document> EnsureEqualsCorrectAsync(Document document,
...
+            {
+                throw new InvalidOperationException("Invocation contains conditional expression but we could not detect the parts.");
+            }
+            else if (conditionalAccessMembers.Count == 1)
+            {
+                return new[] { conditionalAccessMembers[0] }.Concat(arguments);
+            }
+            else
+            {
+                var currentExpression = conditionalAccessMembers.Count > 2
```
Comment (mavasani) [inline src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/Fixers/CompareSymbolsCorrectlyFix.cs:117]: Do you want to file a follow-up issue to simplify this code and directly use Roslyn public API once we move to the new Microsoft.CodeAnalysis in future? Actually, having said that, I think it is about time to move some of analyzer packages to newer Microsoft.CodeAnalysis API, I can do that next week.

**Review by mavasani — APPROVED** [review]

File: src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/Fixers/CompareSymbolsCorrectlyFix.cs (hunk 2 above)
Comment (Evangelink) [reply src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/Fixers/CompareSymbolsCorrectlyFix.cs:117]: https://github.com/dotnet/roslyn-analyzers/issues/4580

**Pull request merged**